shared/version/version.go

```diff
@@ -49,7 +49,7 @@ func NewDottedVersion(versionString string) (*DottedVersion, error) {
 
 // Parse parses a string starting with a dotted version and returns it.
 func Parse(s string) (*DottedVersion, error) {
-	r, _ := regexp.Compile(`^([0-9]+.[0-9]+(.[0-9]+)?).*`)
+	r := regexp.MustCompile(`^([0-9]+.[0-9]+(?:.[0-9]+)?)`)
```

Comment:
We dont panic in lxd, so MustCompile is no good here and should return an error instead.